src/ui/suspense/frameCache.ts

```diff
@@ -60,3 +82,108 @@ function createPauseFrames(pauseId: PauseId, frames: Frame[], sourcesState: Sour
     sourcesState.sourceDetails.entities
   )!;
 }
+
+export interface FormattedPointStackFrame {
+  url?: string;
+  source: Source;
+  functionLocation: Location;
+  executionLocation: Location;
+  point: PointDescription;
+  functionDetails?: FormattedEventListener;
+}
+
+export interface FormattedPointStack {
+  point: PointDescription;
+  resultPoint: PointDescription;
+  frame?: FormattedPointStackFrame;
+  allFrames: FormattedPointStackFrame[];
+  filteredFrames: FormattedPointStackFrame[];
+  functionName?: string;
+}
+
+export async function formatPointStackFrame(
+  replayClient: ReplayClientInterface,
+  frame: PointStackFrame,
+  sourcesById: Map<string, Source>
+): Promise<FormattedPointStackFrame> {
+  updateMappedLocationForPointStackFrame(sourcesById, frame);
+  const functionLocation = getPreferredLocation(sourcesById, [], frame.functionLocation);
+  const executionLocation = getPreferredLocation(sourcesById, [], frame.point.frame ?? []);
+
+  const source = sourcesById.get(functionLocation.sourceId)!;
+
+  return {
+    url: source.url,
+    source,
+    functionLocation,
+    executionLocation,
+    point: frame.point,
+    functionDetails: await formatFunctionDetailsFromLocation(
+      replayClient,
+      "unknown",
+      functionLocation
+    ),
+  };
+}
+
+export const formattedPointStackCache: Cache<
+  [replayClient: ReplayClientInterface, point: TimeStampedPoint],
+  FormattedPointStack
+> = createCache({
+  debugLabel: "FormattedPointStack",
+  getKey: ([replayClient, point]) => point.point,
+  async load([replayClient, point]) {
+    return formatPointStackForPoint(replayClient, point);
+  },
+});
+
+export async function formatPointStackForPoint(
```

Comment:
Exporting this function separately from the cache seems like a foot gun.

You can always call `cache.readAsync` to get the same behavior of the async function, but exporting the function directly seems to risk computing things that don't get cached.

Is there a reason you're exporting them separately?

It looks like the only thing that imports this function is `reactInternalsCaches.ts` and it doesn't use the import.

Reply:
In both these cases I wrote the standalone function first, then wrote the cache later, so the code had been left that way.  I can inline these.